src/libcore/iter/adapters/mod.rs

```diff
@@ -1827,17 +1825,13 @@ where
 
     #[inline]
     fn last(mut self) -> Option<I::Item> {
-        if self.n == 0 {
-            self.iter.last()
-        } else {
-            let next = self.next();
-            if next.is_some() {
-                // recurse. n should be 0.
-                self.last().or(next)
-            } else {
-                None
+        if self.n > 0 {
+            // nth(n) skips n+1
+            if self.iter.nth(self.n - 1).is_none() {
+                return None;
```

Comment:
I like using `Try` in cases like this, `self.iter.nth(self.n - 1)?;`

Reply:
That would work in this case but would make the code less symmetric with the other methods of `Skip` which is why I didn't use `?` here.